pkg/query/endpointset.go

```diff
@@ -598,9 +604,22 @@ func (e *EndpointSet) GetEndpointStatus() []EndpointStatus {
 	defer e.endpointsMtx.RUnlock()
 
 	statuses := make([]EndpointStatus, 0, len(e.endpoints))
+	lblsSet := make([]labels.Labels, 0)
 	for _, v := range e.endpoints {
 		status := v.getStatus()
 		if status != nil {
+			if e.storeSelector.RelabelConfigEnabled() {
```

Comment:
Let's make this implicit inside `MatchStore`. We can check if `relabelConfig` is nil and return `true` in that case.